### DISCUSSION TITLE
Archives being overwritten if generated too quickly

### DISCUSSION THREAD

**Discussion by shsnyder** [operatorai/modelstore#196] [General] [answered]
I am migrating an existing database that is storing models to the modelstore and saw that if I added the models from my database to the modelstore too quickly (within the same second), the versions directory gets updated correctly BUT the timestamped archives get overridden. 

I can resolve this for my use case by putting in a 1 second sleep in my import code to make sure there are no collisions, but the modelstore should detect that models might be overridden and fix it by checking for an existing timestamped versioned entry with the same timestamp and modifying the saved timestamp so that nothing gets overridden.

Since the timestamps are not in UTC time, I imagine that a directory containing the model data being accessed from another server in a different time zone may have the same issue with timestamp collisions.

I guess to resolve this the modelstore would need to :
- Timestamp in UTC time so that all timestamps would only increase over time (so no possibility of TS collision)
- Detect an existing archive with the same (within the same second) timestamp and pause or modify the generated timestamp to avoid a collision
 
Not sure if the new release will resolve this, please let me know if I can help or if there is some configuration setting that would resolve this.

**Comment by nlathia** [comment, marked as answer]
Thanks @shsnyder - posting here to link this thread to the issue you raised, which also links to the pull request that fixes this issue:

- https://github.com/operatorai/modelstore/issues/197

This has been released with `modelstore==0.0.77`
